decorator.rb

```diff
@@ -0,0 +1,27 @@
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def initialize(nameable)
+    super(nameable)
+    @nameable = nameable
+  end
```

Comment:
- [ ] Kindly remove this initialize method because nameable is already initialized in the Decorator class.

⚠️Note: Kindly don't forget the required change in the main message above⚠️